cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -297,6 +297,7 @@ def start_master(etcd):
 
 
 @when('kube-control.connected')
+@when('cdk-addons.configured')
```

Comment:
If I understand right, this will prevent kubernetes-worker from deploying ingress and default-http-backend prematurely. I'm +1 to this :+1: